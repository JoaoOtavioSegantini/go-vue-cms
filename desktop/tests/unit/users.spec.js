import Users from "@/components/Users.vue";
import { mount } from "@vue/test-utils";
import { createStore } from "vuex";

describe("Users.vue", () => {
  const createVuexStore = () => {
    return createStore({
      state() {
        return {
          Users: {
            all: [{ ID: "1", name: "John Doe", email: "johndoe@gmail.com" }],
          },
        };
      },
      mutations: {
        updateUsersList(state) {
          state.Users = [];
        },
      },
      actions: {
        listUsers(ctx) {
          const res = mockGet();
          ctx.commit("updateUsersList", res);
        },
      },
    });
  };

  function factory() {
    const store = createVuexStore();
    return mount(Users, {
      global: {
        plugins: [store],
        mocks: {
          $route: {
            params: {
              id: '1'
            }
          }
        }
      },
    });
  }

  const mockGet = jest.fn();

  jest.mock("axios", () => ({
    get: () => mockGet(),
  }));

  it("renders props when passed", () => {
    const msg = "AÇÕES Usuários Gerenciamento de usuários#nameemail1John Doejohndoe@gmail.comver";

    const wrapper = factory();

    expect(mockGet).toHaveBeenCalled();
    expect(wrapper.text()).toMatch(msg);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
