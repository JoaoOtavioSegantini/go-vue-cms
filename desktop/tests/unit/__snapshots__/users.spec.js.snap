// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users.vue renders props when passed 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col-1 nav\\">
      <p class=\\"display-6\\"><small>AÇÕES</small></p><a href=\\"/users/add\\" class=\\"btn btn-secondary\\" title=\\"Novo usuário\\"><i class=\\"fa fa-plus\\"></i></a>
    </div>
    <div class=\\"col\\">
      <div class=\\"card\\">
        <div class=\\"card-body\\">
          <h5 class=\\"title\\"><i class=\\"fa fa-file-text-o\\"></i> Usuários <small>Gerenciamento de usuários</small></h5>
          <table class=\\"table\\" aria-label=\\"uma tabela de usuários do sistema\\">
            <thead>
              <tr>
                <th scope=\\"col\\">#</th>
                <th scope=\\"col\\">name</th>
                <th scope=\\"col\\">email</th>
                <th scope=\\"col\\"></th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th scope=\\"row\\">1</th>
                <td>John Doe</td>
                <td>
                  <div>johndoe@gmail.com</div>
                </td>
                <td class=\\"text-right\\"><a href=\\"/users/1\\" class=\\"btn btn-primary btn-sm\\">ver</a></td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
